File: src/nova.web/src/types/employee.ts
import { Department } from './department';
export interface Employee {
  employeeId: string;
  firstName: string;
  lastName: string;
  email: string;
  position: string;
  departmentId: string | null;
  managerId: string | null;
  startDate: string;
  status: string;
  createdDate: string;
  modifiedDate: string;

  // Navigation properties (optional)
  department?: Department;
  manager?: Employee;
  directReports?: Employee[];
}

export interface TreeItem {
  id: string;
  children: TreeItem[];
  data: Employee;
  collapsed?: boolean;
}

File: src/nova.core/Entities/Employee.cs
namespace Nova.Core.Entities;
public class Employee
{
    public Guid EmployeeId { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Email { get; set; }
    public required string Position { get; set; }
    public Guid? DepartmentId { get; set; }
    public Guid? ManagerId { get; set; }
    public DateTime StartDate { get; set; }
    public required string Status { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime ModifiedDate { get; set; }

    // Navigation properties
    public Department? Department { get; set; }
    public Employee? Manager { get; set; }
    public ICollection<Employee> DirectReports { get; set; } = new List<Employee>();
}